installutils/helminstall/helm_client.go

```diff
@@ -4,13 +4,26 @@ import (
 	"github.com/solo-io/go-utils/installutils/helminstall/internal"
 	"github.com/solo-io/go-utils/installutils/helminstall/types"
 	"github.com/spf13/afero"
+	"k8s.io/client-go/tools/clientcmd"
 )
 
-// a HelmClient that talks to the kube api server and creates resources
-func DefaultHelmClient() types.HelmClient {
-	return &internal.DefaultHelmClient{
-		Fs:              internal.NewFs(afero.NewOsFs()),
-		ResourceFetcher: internal.NewDefaultResourceFetcher(),
-		HelmLoaders:     internal.NewHelmFactories(),
-	}
+// HelmClient factory that accepts kubeconfig as a file.
+func DefaultHelmClientFileConfig(kubeConfig, kubeContext string) types.HelmClient {
+	return internal.NewHelmClientForFileConfig(
+		internal.NewFs(afero.NewOsFs()),
+		internal.NewDefaultResourceFetcher(),
+		internal.NewHelmFactories(),
+		kubeConfig,
+		kubeContext,
+	)
+}
+
+// HelmClient factory that accepts kubeconfig in memory.
+func DefaultHelmClientMemoryConfig(config clientcmd.ClientConfig) types.HelmClient {
+	return internal.NewHelmClientForMemoryConfig(
+		internal.NewFs(afero.NewOsFs()),
```

Comment:
should the afero filesystem implementation be injectable by the user for cases where we don't have access to read/write on the host file system?